src/extractor/guidance/turn_analysis.cpp

```diff
@@ -143,6 +148,13 @@ Intersection TurnAnalysis::AssignTurnTypes(const NodeID node_prior_to_intersecti
                 road.instruction.type = TurnType::OffRamp;
         });
     }
+    // Suppress turns on ways between mode types that do not need guidance
+    if (suppress_mode_handler.canProcess(
+            node_prior_to_intersection, entering_via_edge, intersection))
+    {
+        intersection = suppress_mode_handler(
+            node_prior_to_intersection, entering_via_edge, std::move(intersection));
+    }
```

Comment:
Also if your handler fires you have to disable all other handlers. Otherwise it could be that e.g. your handler changes instructions to NoTurn and now the Sliproad Handler accidentally is getting confused with the NoTurn and classifies this as a Sliproad.

Make the handler come first and if it triggers skip all other handler.